.gitlab-ci.yml

```diff
@@ -29,22 +29,56 @@ variables:
     paths:
       - artifacts/
 
-test-linux:
+test-audit:
   stage:                           test
-  variables:
-    RUN_TESTS:                     all
   script:
-    - scripts/gitlab/test-all.sh
+    - set -e
+    - set -u
+    - cargo audit
+  tags:
+    - linux-docker
+
+test-validate-chainspecs:
+  stage:                           test
+  script:
+    - ./scripts/gitlab/validate_chainspecs.sh
     - sccache -s
   tags:
     - linux-docker
 
-test-audit:
+cargo-check:
   stage:                           test
   script:
-    - set -e
-    - set -u
-    - cargo audit
+    - ./scripts/gitlab/cargo_check.sh
+    - sccache -s
+  tags:
+    - linux-docker
+
+test-rust-changes:
+  stage:                           test
+  script:
+    - ./scripts/gitlab/rust_changes.sh
+  tags:
+    - linux-docker
+  allow_failure:                   true
+
+test-cpp:
+  stage:                           build
+  dependencies:
+    - test-rust-changes
+  script:
+    - ./scripts/gitlab/test-cpp.sh
+    - sccache -s
+  tags:
+    - linux-docker
+
+test-linux:
+  stage:                           build
+  dependencies:
+    - test-rust-changes
+  script:
+    - ./scripts/gitlab/test.sh
```

Comment:
it's just one command, so why not put it here directly. or if there should be a script with every job, then why not name them in a way that it is obvious which belong together?

Reply:
makes sense to put the command directly here. And will take a look at the naming as well.